endpointmanager/pkg/endpointlinker/endpointlinker_integration_test.go

```diff
@@ -151,6 +152,78 @@ func Test_addMatch(t *testing.T) {
 	th.Assert(t, sConfidence == newConfidence, fmt.Sprintf("expected stored confidence '%f' to be the same as the new confidence that was stored '%f'.", sConfidence, newConfidence))
 }
 
+func Test_manualLinkerCorrections(t *testing.T) {
+	teardown, _ := th.IntegrationDBTestSetup(t, store.DB)
+	defer teardown(t, store.DB)
+
+	ctx := context.Background()
+	ep1 := &endpointmanager.FHIREndpoint{
+		ID:                1,
+		URL:               "example.com/FHIR/DSTU2",
+		OrganizationNames: []string{"FOO FOO BAR"},
+		NPIIDs:            []string{},
+		ListSource:        "https://open.epic.com/MyApps/EndpointsJson"}
+	npiID1 := "1"
+	confidence1 := .6
+	ep2 := &endpointmanager.FHIREndpoint{
+		ID:                2,
+		URL:               "example2.com/FHIR/DSTU2",
+		OrganizationNames: []string{"FOO BAR BAR"},
+		NPIIDs:            []string{},
+		ListSource:        "https://open.epic.com/MyApps/EndpointsJson"}
+	npiID2 := "2"
+	confidence2 := .8
+	ep3 := &endpointmanager.FHIREndpoint{
+		ID:                3,
+		URL:               "example3.com/FHIR/DSTU2",
+		OrganizationNames: []string{"FOO BAR FOO"},
+		NPIIDs:            []string{},
+		ListSource:        "https://open.epic.com/MyApps/EndpointsJson"}
+	npiID3 := "3"
+	confidence3 := .5
+
+	// add matches
+	err := addMatch(ctx, store, npiID1, ep1, confidence1)
+	th.Assert(t, err == nil, err)
+	err = addMatch(ctx, store, npiID2, ep2, confidence2)
+	th.Assert(t, err == nil, err)
+	err = addMatch(ctx, store, npiID3, ep3, confidence3)
+	th.Assert(t, err == nil, err)
+
+	// open fake whitelist and blacklist files
+	whitelistMap, blacklistMap, err := openLinkerCorrectionFiles("../testdata/fakeWhitelist.json", "../testdata/fakeBlacklist.json")
+	th.Assert(t, err == nil, err)
+
+	// run linkerFix manual linker algorithm correction function
+	err = linkerFix(ctx, store, whitelistMap, blacklistMap)
+	th.Assert(t, err == nil, err)
+	ep4URL := "example4.com/FHIR/DSTU2"
+	npiID4 := "4"
+	sNpiID, sEpURL, sConfidence, err := store.GetNPIOrganizationFHIREndpointLink(ctx, npiID4, ep4URL)
+	th.Assert(t, err == nil, err)
+	th.Assert(t, sNpiID == npiID4, fmt.Sprintf("expected stored ID '%s' to be the same as the ID that was stored from whitelist '%s'.", sNpiID, npiID4))
+	th.Assert(t, sEpURL == ep4URL, fmt.Sprintf("expected stored url '%s' to be the same as the url that was stored from whitelist '%s'.", sEpURL, ep4URL))
+	th.Assert(t, sConfidence == 1.000, fmt.Sprintf("expected stored confidence 1.000, got '%f'.", sConfidence))
+
+	sNpiID, sEpURL, sConfidence, err = store.GetNPIOrganizationFHIREndpointLink(ctx, npiID1, ep3.URL)
```

Comment:
Based on how I understood the task, since the link between the 3rd endpoint and Organization 3 is put in the blacklist file, this link should be removed from the endpoint_organization table, so this test checks to see if the link between the 3rd endpoint and Organization 3 was removed correctly

Reply:
ohhh i misread this, I understand now, makes sense!